addon/components/high-charts.js

```diff
@@ -110,7 +110,8 @@ export default Component.extend({
 
   draw() {
     let element = this.element && this.element.querySelector('.chart-container');
-    let mode = CHART_TYPES[get(this, 'mode')];
+    let modeAttr = get(this, 'mode');
+    let mode = CHART_TYPES[!modeAttr ? undefined : modeAttr];
```

Comment:
This is slightly confusing logic. What would you think about

```
let modeAttr = get(this, 'mode') || undefined;
let mode = CHART_TYPES[!modeAttr ? undefined : modeAttr];
```

Reply:
Sorry for my late reply.

I agree that this line makes people read twice:
`let mode = CHART_TYPES[!modeAttr ? undefined : modeAttr];`
If you want I can try to make that more obvious or at least comment it.

I don't see what the ` || undefined ` is supposed to make less confusing.

Reply:
Because if we want to remove the confusing not operator, you need to make a falsey `mode` be `undefined` so that it's correct when you look it up in the mode options hash.

What I **should** have written is

```
let modeAttr = get(this, 'mode') || undefined;
let mode = CHART_TYPES[modeAttr];
```

Reply:
In your defense, I suppose it is quite obvious that was what you meant. I guess my tendency to not assume anything is starting to blindside me.